src/Common.jl

```diff
@@ -51,4 +51,58 @@ function G_func(param_set::APS, T::FT, ::TD.Ice) where {FT <: Real}
     )
 end
 
+"""
+    logistic_function(x, x_0, k)
+ 
+ - `x` - independent variable
+ - `x_0` - threshold value for x
+ - `k` - growth rate of the curve, characterizing steepness of the transition
+
+Returns the value of the logistic function for smooth transitioning at thresholds. This is 
+a normalized curve changing from 0 to 1 while x varies from 0 to Inf (for positive k). For 
+x < 0 the value at x = 0 (zero) is returned. For x_0 = 0 H(x) is returned.
+"""
+function logistic_function(x::FT, x_0::FT, k::FT) where {FT <: Real}
+
+    @assert k > 0
+    @assert x_0 >= 0
+    x = max(0, x)
+
+    # When both x and x_0 are zero t is nan, and need to be replaced by -Inf so that 
+    # the result for x = 0 is zero.
+    _t = (x == 0.0 && x_0 == 0.0) ? -Inf : x / x_0 - x_0 / x
+    _kt = k * _t
+
+    return 1 / (1 + exp(-_kt))
```

Comment:
```suggestion
    if x == 0.0 && x_0 == 0.0
        return FT(0) # avoid computing `exp(-(-Inf))`
    else
        _t = x / x_0 - x_0 / x
        _kt = k * _t
        return 1 / (1 + exp(-_kt))
    end
    _kt = k * _t

    return 1 / (1 + exp(-_kt))
```
I think my suggestion earlier was to use branching (`if`-statements) like this to (sometimes) avoid calling `exp`. It's somewhat minor, and really only performance related.

The more important issue I have is with `x == 0.0 && x_0 == 0.0`. I'm just not sure that this is how we want to do the comparison. Let's consider what happens here:

```julia
julia> compare(num::FT) where {FT} = num == 0.0
compare (generic function with 1 method)

julia> compare(Float32(1e-7))
false

julia> compare(Float32(1e-15))
false

julia> compare(Float32(1e-60))
true

julia> compare(Float32(1e-30))
false

julia> compare(Float64(1e-30))
false

julia> compare(Float64(1e-300))
false

julia> compare(Float64(1e-600))
true

julia> compare(Float64(0.0))
true
```

Reply:
Oh yes you're right. I had this in two functions, fixed one and missed this one. Again here I wanted to catch NaN cases, I guess it would be better to use isnan function instead. But considering performance as you suggested, we can avoid computing exp in some more cases. I'll push new commits.